### ISSUE TITLE
l1 loss suddenly not defined. 

### ISSUE THREAD

**Issue by tunguyenlam** [SayedNadim/Global-and-Local-Attention-Based-Free-Form-Image-Inpainting#11]
Hi,
I have tried to train model but suddenly its l1 loss and generator loss are **nan**. The logs is bellow:
...
020-08-09 11:27:36,361 INFO Iter: [2/500000] l1: 0.057654 g: 1.074609 d: 0.999996 speed: 2.65 batches/s 
2020-08-09 11:27:36,362 INFO saving model into: checkpoints/Places/hole_benchmark
2020-08-09 11:33:50,187 INFO Iter: [2/500000] **l1: nan g: nan** d: 0.920272 speed: 2.68 batches/s 
2020-08-09 11:33:50,187 INFO saving model into: checkpoints/Places/hole_benchmark
2020-08-09 11:40:04,448 INFO Iter: [2/500000] **l1: nan g: nan** d: 0.781143 speed: 2.67 batches/s 
2020-08-09 11:40:04,448 INFO saving model into: checkpoints/Places/hole_benchmark
2020-08-09 11:46:16,912 INFO Iter: [2/500000] **l1: nan g: nan** d: 0.802022 speed: 2.68 batches/s 
...

Do you have any idea about that?

Thanks,

**Comment by SayedNadim** [comment]
Hi,
Can you please tell me about the dataset and your setup?

**Comment by tunguyenlam** [comment]
Hi, 
I tried to test with test data here as its size is small: 
http://data.csail.mit.edu/places/places365/test_large.tar?fireglass_rsn=true

Thanks,

**Comment by SayedNadim** [comment]
Hi, 
Can you please try with the following script for mask?

`
def random_mask(max_mask,
                height=256,
                width=256,
                max_vertex=15,
                max_brush_width=30,
                max_length=200):
    mask = np.zeros((height, width))
    mask_all = []
    max_angle = np.pi
    for mn in range(max_mask):
        num_vertex = int(np.random.uniform(max_vertex//3, max_vertex))
        start_x = int(np.random.uniform(0, height))
        start_y = int(np.random.uniform(0, width))

        for step in range(0, num_vertex):
            angle = np.random.uniform(0, max_angle)

            if step % 2 == 0:
                # Reverse mode
                angle = 2 * math.pi - angle

            length = np.random.uniform(10, max_length)
            brush_width = int(np.random.uniform(max_brush_width / 4, max_brush_width))

            end_x = int(start_x + length * math.sin(angle))
            end_y = int(start_y + length * math.cos(angle))

            cv2.line(mask, (start_y, start_x), (end_y, end_x), 1, brush_width)

            start_x = end_x
            start_y = end_y

            # cv2.circle(mask, (447, 63), brush_width // 4, 255, -1)

        if np.random.uniform(0.0, 1.0) > 0.5:
            mask = np.fliplr(mask)

        if np.random.uniform(0.0, 1.0) > 0.5:
            mask = np.flipud(mask)
        # mask = mask // 255
        mask_all.append(mask)
    batched_mask = torch.tensor(mask_all, dtype=torch.float32).unsqueeze(1)
    return batched_mask
`

**Comment by SayedNadim** [comment]
Hi, can you please try with this mask generation script?
```python

def random_mask(max_mask,
                height=256,
                width=256,
                max_vertex=15,
                max_brush_width=30,
                max_length=200):
    mask = np.zeros((height, width))
    mask_all = []
    max_angle = np.pi
    for mn in range(max_mask):
        num_vertex = int(np.random.uniform(max_vertex//3, max_vertex))
        start_x = int(np.random.uniform(0, height))
        start_y = int(np.random.uniform(0, width))

        for step in range(0, num_vertex):
            angle = np.random.uniform(0, max_angle)

            if step % 2 == 0:
                # Reverse mode
                angle = 2 * math.pi - angle

            length = np.random.uniform(10, max_length)
            brush_width = int(np.random.uniform(max_brush_width / 4, max_brush_width))

            end_x = int(start_x + length * math.sin(angle))
            end_y = int(start_y + length * math.cos(angle))

            cv2.line(mask, (start_y, start_x), (end_y, end_x), 1, brush_width)

            start_x = end_x
            start_y = end_y

            # cv2.circle(mask, (447, 63), brush_width // 4, 255, -1)

        if np.random.uniform(0.0, 1.0) > 0.5:
            mask = np.fliplr(mask)

        if np.random.uniform(0.0, 1.0) > 0.5:
            mask = np.flipud(mask)
        # mask = mask // 255
        mask_all.append(mask)
    batched_mask = torch.tensor(mask_all, dtype=torch.float32).unsqueeze(1)
    return batched_mask
```

**Comment by tunguyenlam** [comment]
thanks @SayedNadim , 
Is it replacement for Masks.get_ff_mask function ? if like that what is max_mask value here. 

Thanks,

**Comment by SayedNadim** [comment]
Hi, the mask value is 0 and 1. Here is the full script. I used it for initial training. Please tell me if this works.
```Python
def random_mask(max_mask,
                height=256,
                width=256,
                max_vertex=15,
                max_brush_width=30,
                max_length=200):
    mask = np.zeros((height, width))
    mask_all = []
    max_angle = np.pi
    for mn in range(max_mask):
        num_vertex = int(np.random.uniform(max_vertex//3, max_vertex))
        start_x = int(np.random.uniform(0, height))
        start_y = int(np.random.uniform(0, width))

        for step in range(0, num_vertex):
            angle = np.random.uniform(0, max_angle)

            if step % 2 == 0:
                # Reverse mode
                angle = 2 * math.pi - angle

            length = np.random.uniform(10, max_length)
            brush_width = int(np.random.uniform(max_brush_width / 4, max_brush_width))

            end_x = int(start_x + length * math.sin(angle))
            end_y = int(start_y + length * math.cos(angle))

            cv2.line(mask, (start_y, start_x), (end_y, end_x), 1, brush_width)

            start_x = end_x
            start_y = end_y

            # cv2.circle(mask, (447, 63), brush_width // 4, 255, -1)

        if np.random.uniform(0.0, 1.0) > 0.5:
            mask = np.fliplr(mask)

        if np.random.uniform(0.0, 1.0) > 0.5:
            mask = np.flipud(mask)
        # mask = mask // 255
        mask_all.append(mask)
    batched_mask = torch.tensor(mask_all, dtype=torch.float32).unsqueeze(1)
    return batched_mask

def mask_image(x, bboxes, config):
    height, width, _ = config['image_shape']
    max_delta_h, max_delta_w = config['max_delta_shape']
    max_mask = x.shape[0]
    mask = random_mask(max_mask)
    if x.is_cuda:
        mask = mask.cuda()

    if config['mask_type'] == 'hole':
        result = x * (1. - mask)
    elif config['mask_type'] == 'mosaic':
        # TODO: Matching the mosaic patch size and the mask size
        mosaic_unit_size = config['mosaic_unit_size']
        downsampled_image = F.interpolate(x, scale_factor=1. / mosaic_unit_size, mode='nearest')
        upsampled_image = F.interpolate(downsampled_image, size=(height, width), mode='nearest')
        result = upsampled_image * mask + x * (1. - mask)
    else:
        raise NotImplementedError('Not implemented mask type.')

    return result, mask
```
